### PR TITLE
Crypto acceleration of SHA-256 and SHA-1

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

**Comment by pascal-brand-st-dev** [comment]
When reading it at first, I thought that It is good to have them as separated algorithm.

However, it is to the cost of a more complex way to manage the defines to ensure the correct version of sha256 is used. So I wonder if it is really important to have both algorithms at the same time.

@jenswi-linaro:  what kind of applications do you foresee where it will be required to have both sha256 and sha256_neon (or sha1 version)

**Comment by jenswi-linaro** [comment]
I don't know, it just seemed like a good idea. Right now I don't see any problems with only having one of the implementations available at a time.

**Comment by jforissier** [comment]
Like @pascal-brand-st-dev I would rather have LTC export only one sha256 (or sha1) version, because it seems it would make the code simpler.

**Comment by jenswi-linaro** [comment]
OK, I'll change to only export one SHA-256 and one SHA-1 version at a time.

**Comment by jenswi-linaro** [comment]
Review comments addressed

**Comment by jenswi-linaro** [comment]
A few more small fixes

**Comment by jenswi-linaro** [comment]
Would it be easier if I squashed and rebased the patches for a second review?

**Comment by pascal-brand-st-dev** [comment]
Yes please

**Comment by jenswi-linaro** [comment]
Please ignore the checkpatch warnings/errors on LTC coding style.

**Comment by jenswi-linaro** [comment]
Renamed functions etc as requested.

File: core/lib/libtomcrypt/src/hashes/sha2/sha256_arm32_ce.c [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+  }
+  return CRYPT_OK;
+ #endif
+}
+
+#endif
+
+
+
+/* $Source: /cvs/libtom/libtomcrypt/src/hashes/sha2/sha256.c,v $ */
```
Comment (pascal-brand-st-dev) [inline core/lib/libtomcrypt/src/hashes/sha2/sha256_arm32_ce.c]: Obsolete comment

File: core/lib/libtomcrypt/src/hashes/sha1_arm32_ce.c [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  }
+  return CRYPT_OK;
+  #endif
+}
+
+#endif
+
+
+
+/* $Source: /cvs/libtom/libtomcrypt/src/hashes/sha1.c,v $ */
```
Comment (pascal-brand-st-dev) [inline core/lib/libtomcrypt/src/hashes/sha1_arm32_ce.c]: Obsolete comment

**Comment by pascal-brand-st-dev** [comment]
Apart 2 minor comments,

    Reviewed-by: Pascal Brand <pascal.brand@linaro.org>
    Tested-by: Pascal Brand <pascal.brand@linaro.org> (STM platform)

**Comment by jenswi-linaro** [comment]
Squashed and tags applied

**Comment by jenswi-linaro** [comment]
Rebased on master

**Pull request merged**